### PR TITLE
Add doc for StatefulSet Creation

### PR DESCRIPTION
**Pull request by sablumiah**
WHAT
Document on what and how to use StatefulSets
WHY
Users may have a requirement to have persistent volumes with pods and using StatefulSets enables PVs to be created dynamically as opposed to admins creating PVs for users

### CODE REVIEW

File: source/documentation/other-topics/statefulsets.md [hunk 1]
```diff
@@ -0,0 +1,122 @@
+### StatefulSets (Pods with Persistent Volumes)
+
+#### Introduction
+
+Like a Deployment, a [StatefulSet](https://kubernetes.io/docs/concepts/workloads/controllers/statefulset/) manages Pods that are based on an identical container spec. Unlike a Deployment, a StatefulSet maintains a sticky identity for each of their Pods. These pods are created from the same spec, but are not interchangeable: each has a persistent identifier (UID) that it maintains across any rescheduling. It also provides guarantees ordering and uniqueness of the Pods.
```
Comment (alkar) [inline source/documentation/other-topics/statefulsets.md]: Perhaps:
"It also provides guarantees _on the_ ordering and [...]"

File: source/documentation/other-topics/statefulsets.md [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+StatefulSets also require a [headless service](https://kubernetes.io/docs/concepts/services-networking/service/#headless-services) to manage network identities for pods. Each Pod gets a sticky network sub-domain and CRV record that does not change if a Pod is deleted and recreated.
```
Comment (alkar) [inline source/documentation/other-topics/statefulsets.md]: Did you mean `SRV` record?

File: source/documentation/other-topics/statefulsets.md (hunk 1 above)
Comment (sablumiah) [reply source/documentation/other-topics/statefulsets.md]: Corrected

File: source/documentation/other-topics/statefulsets.md (hunk 2 above)
Comment (sablumiah) [reply source/documentation/other-topics/statefulsets.md]: corrected

**Pull request merged**